src/com/pty4j/windows/conpty/WinConPtyProcess.java

```diff
@@ -53,20 +53,35 @@ public WinConPtyProcess(@NotNull PtyProcessOptions options, @Nullable LongConsum
     }
     myInputStream = new WinHandleInputStream(outPipe.getReadPipe());
     myOutputStream = new WinHandleOutputStream(inPipe.getWritePipe());
-    startAwaitingThread(List.of(options.getCommand()));
+    startAwaitingThread(options.getCommandLine());
   }
 
+
+  /**
+   * Retrieves the command associated with this process. See {@code CommandLine.toList()} for more details.
+   *
+   * @deprecated May return a processed commandline. Use {@link #getCommandLine()} instead.
+   *
+   * @return a {@link List} of strings representing the command line.
+   */
+  @Deprecated
   public @NotNull List<String> getCommand() {
-    return myCommand;
+    return myCommandLine.toList();
+  }
+
+  @NotNull
+  public CommandLine getCommandLine() {
+    return myCommandLine;
   }
 
   private static @NotNull WinSize getInitialSize(@NotNull PtyProcessOptions options) {
     return new WinSize(Objects.requireNonNullElse(options.getInitialColumns(), 80),
         Objects.requireNonNullElse(options.getInitialRows(), 25));
   }
 
-  private void startAwaitingThread(@NotNull List<String> command) {
-    String commandLine = String.join(" ", command);
+  private void startAwaitingThread(@NotNull CommandLine command) {
+    String commandLine = command.toString();
```

Comment:
Let's rename `command` to `commandLine` and use `commandLine.toCommandLineString()` instead of `toString`?